src/validateWebSite.ts

```diff
@@ -24,78 +23,57 @@ type WebError = {
 };
 
 interface IValidateProperties {
-    statusCodes?: string; // Comma-delimited
-    lastStatusCode?: string;
-    lastStatusMessage?: string;
-    timedOut?: 'true' | 'false';
+    statusCodes?: string; // [[code,elapsedSeconds], [code,elapsedSeconds]...]
 }
 
+const initialPollingIntervalMs = 5000;
+const pollingIncrementMs = 0; // Increase in interval each time
+const maximumValidationMs = 60 * 1000; //
+
 export async function validateWebSite(siteTreeItem: SiteTreeItem, outputChannel: OutputChannel, telemetryReporter: TelemetryReporter): Promise<void> {
-    const siteWrapper = siteTreeItem.siteWrapper;
     return callWithTelemetryAndErrorHandling('appService.validateWebSite', telemetryReporter, outputChannel, async function (this: IActionContext): Promise<void> {
         this.rethrowError = false;
         this.suppressErrorDisplay = true;
 
         const properties = <IValidateProperties>this.properties;
 
-        let pollingIntervalMs = 1000;
-        const pollingIncrementMs = 1000; // Increase in interval each time
-        const timeoutTime = Date.now() + 60 * 1000;
+        let pollingIntervalMs = initialPollingIntervalMs;
+        const start = Date.now();
         const uri = siteTreeItem.defaultHostUri;
         const options: {} = {
             method: 'GET',
             uri: uri,
             resolveWithFullResponse: true
         };
         let currentStatusCode: number = 0;
-        let currentStatusMessage: string = '';
-
-        properties.statusCodes = '';
-        properties.timedOut = 'false';
-
-        log(siteWrapper, outputChannel, `Checking for successful response from ${uri}.`);
+        const statusCodes: { code: number, elapsed: number, reported?: boolean }[] = [];
 
         // tslint:disable-next-line:no-constant-condition
         while (true) {
-            let isSuccess: boolean;
             try {
                 const response = <IncomingMessage>(await requestPromise(options));
-                // request throws an error for 400-500 responses
-                // a status code 200-300 indicates success
                 currentStatusCode = response.statusCode;
-                isSuccess = response.statusCode >= 200 && response.statusCode < 400;
-                currentStatusMessage = response.statusMessage;
-            } catch (e) {
-                const error = <WebError>e;
-                const response = error.response || {};
-                isSuccess = false;
+            } catch (error) {
+                const response = (<WebError>error).response || {};
```

Comment:
nit: Check if error is null or undefined before getting error.response

Reply:
It doesn't seem reasonable that I should have to check on every catch if someone threw null or undefined. You disagree?

Reply:
Well most of the time people use parseError(), which automatically checks for them

And yes - I'm very strict/cautious about error handling and telemetry. I just think the amount of work we have to do is worth the amount of frustration from a user if they got a null ref exception because of telemetry

But on the other hand I recognize its a bit of a philosophical debate since you're unlikely to get null or undefined here. So its not something I'd block a PR for